File: Jack.Business/ICrud.vb
Public Interface ICrud(Of TClass, keyType)

    Function Insert(ByVal oTipo As TClass) As Boolean
    Function Update(ByVal oTipo As TClass) As Boolean
    Function Delete(ByVal oTipo As TClass) As Boolean
    Function Find(ByVal Identifier As keyType) As TClass
    Function LoadAll() As IList(Of TClass)

End Interface

File: Jack.Application/ICrud.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Jack.Application
{
    public interface ICrud<TClass, keyType>
    {

        bool Insert(TClass oTipo);
        bool Update(TClass oTipo);
        bool Delete(TClass oTipo);
        TClass Find(keyType Identifier);
        IList<TClass> LoadAll();

    }
}
